test/get_client_stats_test.dart

```diff
@@ -13,13 +14,13 @@ void main() {
       var response = await index.addDocuments([
         {'book_id': 123, 'title': 'Pride and Prejudice'},
         {'book_id': 456, 'title': 'The Martin'},
-      ]).waitFor();
+      ]).waitFor(client: client);
```

Comment:
why is client suddenly needed? This is just to understand the code, nothing to do with the PR 😊

Reply:
If you go to `test/utils/client.dart` you'll see a `late MeilisearchClient client` variable definition. This variable is basically a global variable that is available to everyone who import the file  `test/utils/client.dart` in their tests.

That's hard to reuse because I moved the `waitFor` extensions to another file and suddenly the waitFor functions did not work anymore since they relied on that global variable. My whole point with this refactor was to remove the dependency of that client, so it was pointless to import the same problematic file `tests/utils/client.dart`.
So, I decided to take it by argument, and it solved the issue, but indeed it was not required.